### ISSUE TITLE
MacBook does not always get DHCP lease from airbase-ng softAP

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#760]
**Reported by misterx on 26 Jun 2010 20:00 UTC**

In most cases, PC and mobile phones get the DHCP lease correctly, but it sometimes does not work with macbook pros.

**Comment by aircrack-ng** [comment]
**Comment by misterx on 26 Jun 2010 20:00 UTC**

Might be due to the capabilities in the beacons that aren't correctly set.

**Comment by aircrack-ng** [comment]
**Comment by misterx on 26 Jun 2010 20:00 UTC**

or maybe in the probe response.

**Comment by aircrack-ng** [comment]
**Comment by misterx on 26 Jun 2010 20:00 UTC**

URL: http://backtrack-notes.blogspot.com/2010/08/patch-airbase-ngc.html

Fix:

```
    pthread_mutex_unlock( &mx_cap );

    z = ( ( packet[& 3 ) != 3 ) ? 24 : 30;
+   if(packet[0](1]) == 0x88)
+       z += 2; /* handle QoS field */

    if(length < z)
    {
```

**Comment by aircrack-ng** [comment]
**Comment by misterx on 26 Jun 2010 20:00 UTC**

Fixed in aircrack-ng/aircrack-ng@00abfc0981d45b15e9e39b6867923306be74c50c.
